### PR TITLE
Add new API to loggers endpoint: support delete logger configuration.

### LINKED ISSUE #8776: LoggersMvcEndpoint add the ability to delete logger configuration.
After I add new logging configuration I can't remove it without restarting the component.

### PR DESCRIPTION
**Pull request by igreenfield**
Able to remove logger configuration from the loggers endpoint  fix #8776

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@igreenfield Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8830)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8830) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@igreenfield Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8830)!

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/ActuatorDeleteMapping.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/**
+ * Specialized {@link RequestMapping} for {@link RequestMethod#DELETE DELETE} requests that
+ *
+ * @author Izek Greenfield
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@RequestMapping(method = RequestMethod.DELETE)
+@interface ActuatorDeleteMapping {
```
Comment (wilkinsona) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/mvc/ActuatorDeleteMapping.java]: This doesn't seem to be adding much value as there's no content type. As such, it's exactly the same as the standard `@DeleteMapping` in Spring Framework and I think it'd be better to just use that instead.

File: spring-boot/src/main/java/org/springframework/boot/logging/LoggingSystem.java [hunk 2]
```diff
@@ -143,6 +143,15 @@ public LoggerConfiguration getLoggerConfiguration(String loggerName) {
 	}
 
 	/**
+	 * Delete configuration for a {@link LoggingSystem}'s logger.
+	 * @param loggerName the name of the logger
+	 * @since 1.5.0
+	 */
+	public void deleteLoggerConfiguration(String loggerName) {
+		throw new UnsupportedOperationException("Unable to delete logger configuration");
```
Comment (wilkinsona) [inline spring-boot/src/main/java/org/springframework/boot/logging/LoggingSystem.java]: The actual implementation of deleting a logger seems to be missing. The various `LoggingSystem` subclasses need to be updated to override this method with a logging system-specific implementation.

File: spring-boot/src/main/java/org/springframework/boot/logging/LoggingSystem.java [hunk 3]
```diff
@@ -143,6 +143,15 @@ public LoggerConfiguration getLoggerConfiguration(String loggerName) {
 	}
 
 	/**
+	 * Delete configuration for a {@link LoggingSystem}'s logger.
+	 * @param loggerName the name of the logger
+	 * @since 1.5.0
```
Comment (wilkinsona) [inline spring-boot/src/main/java/org/springframework/boot/logging/LoggingSystem.java]: 1.5.0 has shipped. I think it's most likely that this'll go in 2.0.

**Comment by snicoll** [comment]
@igreenfield can you please rework your PR with the comments above? You can push your changes to the existing branch and i'll update the commit.

**Comment by igreenfield** [comment]
I will, in few days...

On Apr 14, 2017 10:49, Stéphane Nicoll <notifications@github.com> wrote:

@igreenfield<https://github.com/igreenfield> can you please rework your PR with the comments above? You can push your changes to the existing branch and i'll update the commit.

**Comment by igreenfield** [comment]
@snicoll @wilkinsona Please rereview.

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/LoggersEndpoint.java [hunk 4]
```diff
@@ -91,6 +91,11 @@ public void setLogLevel(String name, LogLevel level) {
 		this.loggingSystem.setLogLevel(name, level);
 	}
 
+	public void deleteLoggerConfiguration(String name) {
+		Assert.notNull(name, "Name must not be empty");
```
Comment (snicoll) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/LoggersEndpoint.java]: `null` vs. `empty` here.

Comment (snicoll) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/LoggersEndpoint.java]: `null` vs. `empty` here.

Comment (igreenfield) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/LoggersEndpoint.java]: Do you want me to change the Message to null?

**Comment by igreenfield** [comment]
@snicoll Do you know why the compilation error? when I run mvn package it ok.

**Comment by igreenfield** [comment]
@philwebb What the status of this?

**Comment by snicoll** [comment]
@igreenfield no need to mention anyone. We're all watching the tracker. You can see that @philwebb assigned M2 to this issue (see just above your comment or on the right hand side). It means a version has been assigned to this PR and we'll handle it when that makes sense.

**Comment by igreenfield** [comment]
OK thanks!!!

**Comment by snicoll** [comment]
@igreenfield thanks for the PR but you should be able to "reset" the log level for a particular logger by passing `null`. There are a few things to consider to make that happen (and I'll do that in #8776).

deleting the resource and then expecting the resource to be still available is wrong from an HTTP standpoint anyway.

**Comment by igreenfield** [comment]
@snicoll What do you mean passing null? and where?

**Comment by snicoll** [comment]
Like I said, I am working on it in #8776

**Pull request closed without merging**